packages/rest/src/lib/REST.ts

```diff
@@ -110,6 +137,20 @@ export class REST extends EventEmitter {
 		this.requestManager = new RequestManager(options)
 			.on(RESTEvents.Debug, this.emit.bind(this, RESTEvents.Debug))
 			.on(RESTEvents.RateLimited, this.emit.bind(this, RESTEvents.RateLimited));
+
+		const emitInvalid = this.emit.bind(this, RESTEvents.InvalidRequestWarning);
+
+		this.on('newListener', (eventName) => {
+			if (eventName !== RESTEvents.InvalidRequestWarning) return;
+			if (this.requestManager.listenerCount(RESTEvents.InvalidRequestWarning)) return;
+			this.requestManager.on(RESTEvents.InvalidRequestWarning, emitInvalid);
+		});
+
+		this.on('removeListener', (eventName) => {
+			if (eventName !== RESTEvents.InvalidRequestWarning) return;
+			if (this.listenerCount(RESTEvents.InvalidRequestWarning)) return;
+			this.requestManager.off(RESTEvents.InvalidRequestWarning, emitInvalid);
```

Comment:
but why? We don't do this anywhere else

Reply:
The event count was being checked, but I ran a test suite and discovered checking the event count is far less performant than just emitting, so I'll remove this and its counterpart.